test/e2e/local-cluster/create_cluster.go

```diff
@@ -0,0 +1,229 @@
+
+}
+
+func main() {
+
+	ctx := context.Background()
+	l := log.FromContext(ctx)
+
+	// Create cluster
+	cluster := kind.NewCluster(kindClusterName)
+	cluster.SetDefaults()
+	kubeconfig, err := cluster.Create(ctx)
+	if err != nil {
+		l.Error(err, "cluster creation")
+		os.Exit(1)
+	}
+	l.Info("cluster created successfully")
+
+	// Export kubeconfig
+	f, err := os.Create(kubeconfigName)
+	if err != nil {
+		l.Error(err, "kubeconfig creation")
+		os.Exit(1)
+	}
+	args := []string{"export", "kubeconfig", "--name", kindClusterName, "--kubeconfig", f.Name()}
+	cmd := exec.Command("kind", args...)
+	_, err = cmd.Output()
+	if err != nil {
+		l.Error(err, "kubeconfig export")
+		os.Exit(1)
+	}
+	f, err = os.Create(kubeconfigInternalName)
+	if err != nil {
+		l.Error(err, "kubeconfig creation")
+		os.Exit(1)
+	}
+	args = []string{"export", "kubeconfig", "--name", kindClusterName, "--internal", "--kubeconfig", f.Name()}
+	cmd = exec.Command("kind", args...)
+	_, err = cmd.Output()
+	if err != nil {
+		l.Error(err, "kubeconfig export")
+		os.Exit(1)
+	}
+
+	l.Info("kubeconfig exported successfully")
+
+	// Build image
+	image := fmt.Sprintf("%s:%s", dockerImageRepository, dockerImageTag)
+	if !dockerImageBuildSkip {
+		err = dockerImageBuild("./../../../", image)
+		if err != nil {
+			l.Error(err, "image build")
+			os.Exit(1)
+		}
+		l.Info("Docker image built successfully")
+	}
+
+	// Load image
+	err = cluster.LoadImage(ctx, image)
+	if err != nil {
+		l.Error(err, "image load")
+		os.Exit(1)
+	}
+	l.Info("Docker image loaded to the cluster successfully")
+
+	// Deploy Greenhouse manager
+	err = installChart(ctx, "./../../../charts/manager", greenhouseControllerManagerRelease, kubeconfig, greenhouseControllerManagerNamespace)
+	if err != nil {
+		l.Error(err, "deploy greenhouse")
+		os.Exit(1)
+	}
+	l.Info("Greenhouse manager is deployed successfully")
+
+}
+
+func installChart(ctx context.Context, dir, release, kubeconfig string, namespace string) error {
+
+	l := log.FromContext(ctx)
+
+	chart, err := loader.Load(dir)
+	if err != nil {
+		return err
+	}
+
+	actionConfig := new(action.Configuration)
+	if err := actionConfig.Init(
+		&genericclioptions.ConfigFlags{
+			KubeConfig: &kubeconfig,
+			Namespace:  &namespace,
+		},
+		namespace,
+		"secret",
+		l.V(10).Info,
+	); err != nil {
+		return err
+	}
+
+	// These values are the minimal set of required values to deploy manager helm chart
+	globalDNS := map[string]interface{}{"dnsDomain": "greenhouse.cloudoperators"}
+	chart.Values["global"] = globalDNS
+
+	controllerManagerImage := map[string]interface{}{"repository": dockerImageRepository, "tag": dockerImageTag}
+	controllerManager, ok := chart.Values["controllerManager"].(map[string]interface{})
+	if !ok {
+		l.Error(err, "value merge")
+		os.Exit(1)
+	}
+	controllerManager["image"] = controllerManagerImage
+	controllerManager["replicas"] = "1"
+	chart.Values["controllerManager"] = controllerManager
```

Comment:
Can we somehow load this config from a values file instead of hardcoding?
E.g.:

```golang
var (
	...

	valuesFileName  = flag.String("valuesFile", "./specs/manager.values.yaml", "path to the values file")
	values          = map[string]interface{}{}
	
    ...
)
....

valuesFile, err := os.ReadFile(*valuesFileName)
if err != nil {
	return ctx, err
}
err = yaml.Unmarshal(valuesFile, &values)
if err != nil {
	return ctx, err
}

....

_, err = installAction.RunWithContext(ctx, chart, values)
```

Reply:
done with cd9b6279c0847fdd58c0d4105dd58d539a8d293c